### PR TITLE
fix: update table for mobile view

### LINKED ISSUE #2662: Crypto-to-fiat: Drains history table UI fixes
## Description

This issue covers the transaction table that requires some minor adjustments to align it with other table styling. I haven't been able to test the default state yet, so more issues my come with this. . 

- [ ] The status arrow should show by default, it's currently not showing, and the user do not know it's available for sorting. The table should stay by default sorted by date, so no change here. 

image

- [ ] Mobile table styling should match Figma designs on mobile

**Current**
Screenshot 2024-07-29 at 12 24 20
**Figma**
Screenshot 2024-07-29 at 13 57 19

### PR DESCRIPTION
**Pull request by Nortsova**
## Description

- [x] The status arrow should show by default, it's currently not showing, and the user do not know it's available for sorting. The table should stay by default sorted by date, so no change here. 

image

- [x] Mobile table styling should match Figma designs on mobile figma: https://www.figma.com/design/C2grfQysdsYXz0j4rADR6K/Crypto-to-Fiat?node-id=4456-37629&node-type=frame&t=jlSKOZ8yTQr98oqZ-0

| **Before** | **Now** |
| ----- | ----- |
| Screenshot 2024-07-29 at 12 24 20 | image |

## Testing
1. Create `.diff` file in your root direction
2. Insert this changes:
```
diff --git a/src/components/frame/v5/pages/UserCryptoToFiatPage/partials/FiatTransfersTable/FiatTransfersTable.tsx b/src/components/frame/v5/pages/UserCryptoToFiatPage/partials/FiatTransfersTable/FiatTransfersTable.tsx
index 6108cd1dc..167bc241a 100644
--- a/src/components/frame/v5/pages/UserCryptoToFiatPage/partials/FiatTransfersTable/FiatTransfersTable.tsx
+++ b/src/components/frame/v5/pages/UserCryptoToFiatPage/partials/FiatTransfersTable/FiatTransfersTable.tsx
@@ -45,7 +45,58 @@ const FiatTransfersTable = () => {
   ]);
 
   const isMobile = useMobile();
-  const { sortedData, loading } = useFiatTransfersData(sorting);
+  const { sortedData: data, loading } = useFiatTransfersData(sorting);
+
+  const sortedData: typeof data = [
+    {
+      __typename: 'BridgeDrain',
+      id: '12343',
+      amount: '1234.22',
+      currency: 'usdc',
+      state: 'payment_processed',
+      createdAt: '2024-09-20T13:11:22.310Z',
+      receipt: {
+        __typename: 'BridgeDrainReceipt',
+        url: '#',
+      },
+    },
+    {
+      __typename: 'BridgeDrain',
+      id: '12344',
+      amount: '24689.00',
+      currency: 'usdc',
+      state: 'in_review',
+      createdAt: '2024-08-17T10:14:08.450Z',
+      receipt: {
+        __typename: 'BridgeDrainReceipt',
+        url: '#',
+      },
+    },
+    {
+      __typename: 'BridgeDrain',
+      id: '12345',
+      amount: '2230.77',
+      currency: 'usdc',
+      state: 'awaiting_funds',
+      createdAt: '2024-07-16T06:04:23.100Z',
+      receipt: {
+        __typename: 'BridgeDrainReceipt',
+        url: '#',
+      },
+    },
+    {
+      __typename: 'BridgeDrain',
+      id: '12346',
+      amount: '1200662.77',
+      currency: 'usdc',
+      state: 'awaiting_funds',
+      createdAt: '2024-07-16T06:04:23.100Z',
+      receipt: {
+        __typename: 'BridgeDrainReceipt',
+        url: '#',
+      },
+    },
+  ];
 
   const columns = useFiatTransfersTableColumns(loading);
 
```
3. `git apply .diff`
4. Navigate to Crypto to fiat tab http://localhost:9091/account/crypto-to-fiat 
5. Check that the Drain history matches the design
6. Check that the Status column has an arrow.

Resolves #2662

### CODE REVIEW

**Review by rdig — CHANGES_REQUESTED** [review]
Table itself is all good, but shouldn't it have more padding at the bottom ?

As it stands, on  my machine at least, it renders glued to the bottom of the viewport

Screenshot from 2024-10-21 14-43-23
Screenshot from 2024-10-21 14-43-32
Screenshot from 2024-10-21 14-43-58
Screenshot from 2024-10-21 14-44-06

**Comment by Nortsova** [comment]
Thank you @rdig . I made some changes and now this table should match on both, desktop and mobile. I also added padding-bottom 🙌 

image

image

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Great work on this so far, the styling is spot on:

Screenshot 2024-10-23 at 12 46 34

I'm rejecting because the status sorting doesn't actually work until you sort by one of the other columns first:

https://github.com/user-attachments/assets/f2663a28-d382-4bfe-8fcf-1563eb2fb56d

It is also possible to sort by amount, so I'm not sure if that should be restricted so that it is not possible or whether the arrow needs to be visible here as the original issue states the reason for adding the arrow to the state field is to show that it can be sorted.

**Comment by Nortsova** [comment]
Hey Sam! Nice catch. I think we don't have this design before with multiple arrows, I refactored a bit and now it should works fine:
sorting

I will message @melyndav regarding the Amount sorting! thank you :)

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
This is looking closer now! Unfortunately I'm experiencing this weird issue with the data not sorting and the arrow direction changing on other columns. I'm not exactly sure how this is supposed to function with multiple visible arrows.

https://github.com/user-attachments/assets/c8891733-9bd4-4e26-97ac-ad72937bff93

**Comment by Nortsova** [comment]
Thank you @iamsamgibbs for your attantion to details, I spoke to @melyndav today and it seems I misunderstood the design. 

Here is updated version: 
drain-table

Arrows will be hidden for all columns and visible for sorted ones, also on hover, you can see an arrow that means that the column is sortable 🙌

File: src/components/v5/common/Table/Table.tsx [hunk 1]
```diff
@@ -333,7 +333,7 @@ const Table = <T,>({
                         {row.getVisibleCells().map((cell, index) => {
                           const renderCellWrapperCommonArgs = [
                             clsx(
-                              'flex h-full flex-col items-start justify-center p-[1.1rem] text-md',
+                              'flex h-full flex-col items-start justify-center p-4 text-md',
```
Comment (iamsamgibbs) [inline src/components/v5/common/Table/Table.tsx:336]: Does this change break anything for tables elsewhere in the app? I couldn't check the activity feeds as previous actions weren't loading for me at the time of testing.

**Review by iamsamgibbs — APPROVED** [review]
Great work @Nortsova - this is working really nicely and makes a lot more sense now.

https://github.com/user-attachments/assets/4ea6ba20-5c2c-4a76-8973-c372f412c7cd

And working on mobile too:

https://github.com/user-attachments/assets/868a98e9-a2be-45c0-8146-1e8a8b37d17b

Code changes look sensible and nice to see that you got the arrow hover working too!

**Review by rdig — APPROVED** [review]
All good here! Very nice!

Screenshot from 2024-11-01 17-19-55
Screenshot from 2024-11-01 17-20-07
Screenshot from 2024-11-01 17-20-28
Screenshot from 2024-11-01 17-20-47

File: src/components/frame/v5/pages/UserCryptoToFiatPage/partials/FiatTransfersTable/FiatTransferDescriptionRow.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+interface FiatTransferDescriptionRowProps {
+  title: string;
+  loading: boolean;
+}
+
+export const FiatTransferDescriptionRow: FC<
+  PropsWithChildren<FiatTransferDescriptionRowProps>
+> = ({ title, children, loading }) => {
+  const textClassName = 'font-normal text-sm flex items-center';
```
Comment (bassgeta) [inline src/components/frame/v5/pages/UserCryptoToFiatPage/partials/FiatTransfersTable/FiatTransferDescriptionRow.tsx:12]: I guess we can either wrap this in `tw` or just smash it down into the className directly

**Review by bassgeta — APPROVED** [review]
Yup looks legit and also works as it should :+1: 
Mobile:
image
image
Desktop:
image

**Pull request merged**